pkg/operators/combiner/combiner.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package combiner is a data operator that combines data from a same data
+// source coming from different targets into a single data source. This is
+// useful when we run the same gadget in a distributed environment like
+// Kubernetes and we want to combine the data from all the nodes on the client
+// side so that we can perform further operations on the combined data, e.g.
+// sorting. Notice that this operator is useful only when we have data sources
+// of type array.
+package combiner
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+const (
+	// We need it to be the first operator to run in client side. Only operators
+	// registering data sources should run before the combiner operator.
+	Priority             = -500
+	DataSourcePrefix     = "combined"
+	OperatorName         = "Combiner"
+	OperatorInstanceName = "CombinerInstance"
```

Comment:
Please just use one name for Operator + Instance (both "Combiner" or "combiner") - this is partially used for param creation, so it's less confusing if they're the same.

Reply:
We need to do this in other operators then.

Reply:
Most of them already do it that way, but yeah, there are a couple leftovers that need fixing.